src/runtime/storage/icorrelation_repository.ts

```diff
@@ -91,11 +92,39 @@ export interface ICorrelationRepository {
    */
   getSubprocessesForProcessInstance(processInstanceId: string): Promise<Array<CorrelationFromRepository>>;
 
+  /**
+   * Returns a list of all Correlations in the specified state.
+   *
+   * @async
+   * @param state   state The state by which to retrieve the correlations.
+   * @returns             The retrieved correlations.
+   */
+  getCorrelationsByState(state: CorrelationState): Promise<Array<CorrelationFromRepository>>;
+
   /**
    * Removes all correlations with a specific ProcessModelId.
    *
    * @async
    * @param   processModelId The ID of the processModel, by which correlations should be removed.
    */
   deleteCorrelationByProcessModelId(correlationId: string): Promise<void>;
+
+  /**
+   * Finishes the given correlation.
+   *
+   * @async
+   * @param correlationId   The ID of the Correlation to finish.
```

Comment:
```suggestion
   * @param  correlationId  The ID of the Correlation to finish.
```

Just to keep it aligned with the `@throws` comment.